proxylib/proxylib/connection.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package proxylib
+
+import (
+	"time"
+
+	"github.com/cilium/cilium/pkg/envoy/cilium"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// A parser sees data from the underlying stream in both directions
+// (original, connection open direction and the opposite, the reply
+// direction). Each call to the filter returns an ordered set of
+// operations to be performed on the data in that direction. Any data
+// left over after the returned operations must be buffered by the
+// caller and passed in again when more data has been received on the
+// connection.
+
+type Direction struct {
+	InjectBuf *[]byte
```

Comment:
Would be great to define a type for `*[]byte` if possible

Reply:
I defined the type `InjectBuf` and used that instead of `Direction`, as that was the only thing 
in there. Still have to use `*[]byte` in the exported API function prototype `OnNewConnection` due to cgo restrictions.